### PR TITLE
[IE CLDNN] Export runtime graph as nGraph function

### PR DESCRIPTION
**Pull request by mletavin**
This patch introduces NGraph function as new default format for execution graph export in GPU plugin.

### CODE REVIEW

File: inference-engine/src/cldnn_engine/cldnn_graph.cpp [hunk 1]
```diff
@@ -252,34 +316,193 @@ InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(s
...
+        create_ngraph_node(pi);        
+    }
+
+    ngraph::op::GenericIE::DisableReshape reshape(nodes);
+    auto function = std::make_shared<ngraph::Function>(results, params, "runtime_gpu_graph");
+    InferenceEngine::CNNNetwork net(function);
+    return net;
+}
+
+InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(std::vector<cldnn::primitive_info>& primitives_info,
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_graph.cpp:477]: Why ngraph version can't be used for intermediate stages?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/exec_graph_info.cpp [hunk 2]
```diff
@@ -15,12 +15,21 @@ namespace {
...
                             ExecGraphTests::getTestCaseName);
-
-}  // namespace
\ No newline at end of file
+    INSTANTIATE_TEST_CASE_P(smoke_Multi_BehaviorTests, ExecGraphTests,
+                            ::testing::Combine(
+                                ::testing::ValuesIn(netPrecisions),
+                                ::testing::Values(CommonTestUtils::DEVICE_MULTI),
+                                ::testing::ValuesIn(multiConfigs)),
+                            ExecGraphTests::getTestCaseName);
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/exec_graph_info.cpp:34]: how does it work? exec graph for MULTI case? which graph CPU or GPU is shown?

File: inference-engine/src/cldnn_engine/cldnn_graph.cpp [hunk 3]
```diff
@@ -252,34 +316,193 @@ InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(s
...
+                    port++;
+                }
             }
         }
 
-        layer->params[ExecGraphInfoSerialization::PERF_COUNTER] = exec_time;
-        layer->params[ExecGraphInfoSerialization::OUTPUT_LAYOUTS] = prim_info.layout_str;
-        layer->params[ExecGraphInfoSerialization::EXECUTION_ORDER] = std::to_string(prim_info.exec_id);
+        for (auto&& kvp : extract_primitive_metadata(prim_info)) {
+            if (kvp.first == "layerName") {
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_graph.cpp:438]: Can't it be just read from prim_info without introducing any new property?

File: inference-engine/src/cldnn_engine/cldnn_graph.cpp (hunk 1 above)
Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_graph.cpp]: Fixed - now only the nGraph implementation exists.

File: inference-engine/src/cldnn_engine/cldnn_graph.cpp (hunk 3 above)
Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_graph.cpp]: Fixed.

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/exec_graph_info.cpp (hunk 2 above)
Comment (mletavin) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/exec_graph_info.cpp]: Well, it passes OK.
But, looking closely, this happens to be the exact copy of CPU plugin test functionality, so I'll remove it.

**Pull request merged**